### PR TITLE
Fix expired NOSP behaviour

### PR DESCRIPTION
**Pull request by antonyoneill**
Addressing:
https://hackney.atlassian.net/secure/RapidBoard.jspa?rapidView=30&projectKey=MAAP&modal=detail&selectedIssue=MAAP-146

# What
Fix logic for rules to recommend next action of sending breach notices if NOSP has expired and court date was in past and tenant is still in arrears.

Add test to ensure that a new NOSP to be served is recommended if NOSP has expired without a court date.

### CODE REVIEW

**Comment by antonyoneill** [comment]
Hey, thanks for the comments - part of the problem is I'm not 100% clear on the acceptance. 

This PR is just draft and definitely has some work to do, not intended for eyes without discussion with me :joy:

File: .gitignore [hunk 1]
```diff
@@ -41,3 +41,9 @@ Sessionx.vim
 tags
 # Persistent undo
 [._]*.un~
+
+# Idea
+.idea
```
[the remark this answers is not in the document]
Comment (antonyoneill) [reply .gitignore:46]: While true, I'm very used to these being defined for the project - you can see above where vim temporary files are ignored too.

By placing them in the project it reduces the steps for future developers to take to if they're using a Jetbrains IDE

File: spec/lib/hackney/income/tenancy_classification_spec.rb [hunk 2]
```diff
@@ -104,6 +104,38 @@
...
   end
 
+  context 'when there are arrears' do
+    let(:balance) { 20.00 }
+
+    context 'when a NOSP has expired' do
+      let(:nosp) { Hackney::Domain::Nosp.new(served_date: Time.zone.now - 29.days) }
+
+      context 'without a court date' do
+        let(:courtdate) { nil }
```
[the remark this answers is not in the document]
Comment (antonyoneill) [reply spec/lib/hackney/income/tenancy_classification_spec.rb:114]: I need to talk through this with someone - I expect it to magically be pulled through into the execute function in `Hackney::Income::TenancyClassification`

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (antonyoneill) [reply .gitignore:46]: I'm now reading up on this and the internet seems to be split into three trains of thought:
1. The .idea folder belongs to the project and should be source controlled (!!!) -- I don't think this is right
2. It should be ignored in the project
3. Developers own environmental concerns should not be included within the project (such as `.DS_Store` etc..)

[the remark this answers is not in the document]
Comment (antonyoneill) [reply .gitignore:46]: I'll remove the .idea entry - any ideas where the `.generators` file came from? Is that a normal ruby generated file? Or do you think it's RubyMine related? 

Content:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<Settings><!--This file was automatically generated by Ruby plugin.
You are allowed to: 
1. Reorder generators
2. Remove generators
3. Add installed generators
To add new installed generators automatically delete this file and reload the project.
--><GeneratorsGroup><Generator name="active_record:migration" /><Generator name="active_record:model" /><Generator name="active_record:observer" /><Generator name="active_record:session_migration" /><Generator name="controller" /><Generator name="erb:controller" /><Generator name="erb:mailer" /><Generator name="erb:scaffold" /><Generator name="generator" /><Generator name="helper" /><Generator name="integration_test" /><Generator name="mailer" /><Generator name="metal" /><Generator name="migration" /><Generator name="model" /><Generator name="model_subclass" /><Generator name="observer" /><Generator name="performance_test" /><Generator name="plugin" /><Generator name="resource" /><Generator name="scaffold" /><Generator name="scaffold_controller" /><Generator name="session_migration" /><Generator name="stylesheets" /><Generator name="test_unit:controller" /><Generator name="test_unit:helper" /><Generator name="test_unit:integration" /><Generator name="test_unit:mailer" /><Generator name="test_unit:model" /><Generator name="test_unit:observer" /><Generator name="test_unit:performance" /><Generator name="test_unit:plugin" /><Generator name="test_unit:scaffold" /></GeneratorsGroup></Settings>
```

**Comment by antonyoneill** [comment]
I think it would also be useful to agree on a standard PR template to ensure that we capture the correct information to provide context to reviewers

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (C-gyorfi) [reply .gitignore:46]: Never heard about `.generators` before but based on my googling it must be RubyMine :) Also, feel free keep them here, it was just a suggestion as I like to ignore these in my global `.gitignore` so it fixed for every project

[the remark this answers is not in the document]
Comment (antonyoneill) [reply .gitignore:46]: Implemented the ignore for idea in my global config

https://github.com/antonyoneill/dotfiles/commit/69486c30cf2e25deaac69bde30b908e09f36a59d

**Pull request closed without merging**